src/shared/extensionUtilities.ts

```diff
@@ -369,42 +369,108 @@ export function getComputeRegion(): string | undefined {
-     * @param activityEvents The events that trigger the user activity.
+     * @param customEvents For testing purposes. The events that trigger the user activity.
      * @returns
      */
-    static instance(delay: number = 500, activityEvents = ExtensionUserActivity.activityEvents): ExtensionUserActivity {
-        if (!ExtensionUserActivity.instances[delay]) {
-            const newInstance = new ExtensionUserActivity(delay, activityEvents)
-            ExtensionUserActivity.instances[delay] = newInstance
+    constructor(delay: number = 10_000, customEvents?: vscode.Event<any>[]) {
+        // This ensures we don't fire a user activity event more than once per delay.
+        const throttledEmit = _.throttle(
+            (event: vscode.Event<any>) => {
+                this.activityEvent.fire()
+                getLogger().debug(`ExtensionUserActivity: event fired "${event.name}"`)
+            },
+            delay,
+            { leading: true, trailing: false }
+        )
+
+        if (customEvents) {
+            customEvents.forEach(event =>
+                this.register(event(() => {
+                    throttledEmit(event)
+                }))
+            )
         }
-        return ExtensionUserActivity.instances[delay]
-    }
+        else {
+            this.activityEvents.forEach(event =>
+                this.register(event(() => {
+                    throttledEmit(event)
+                }))
+            )
 
-    private constructor(delay: number, private readonly activityEvents: vscode.Event<any>[]) {
-        const throttledEvent = _.throttle(() => this.activityEvent.fire(), delay, { leading: true })
-        this.activityEvents.forEach(event => event(throttledEvent))
+            this.registerEventsWithEdgeCases(throttledEmit)            
+        }
+
+        this.disposables.push(this.activityEvent)
     }
 
-    private static activityEvents = [
+    private activityEvents: vscode.Event<any>[] = [
+        vscode.window.onDidChangeActiveColorTheme,
         vscode.window.onDidChangeActiveTextEditor,
-        vscode.window.onDidChangeTextEditorSelection,
         vscode.window.onDidChangeActiveTerminal,
         vscode.window.onDidChangeVisibleTextEditors,
-        vscode.window.onDidChangeWindowState,
         vscode.window.onDidChangeTextEditorOptions,
         vscode.window.onDidOpenTerminal,
         vscode.window.onDidCloseTerminal,
-        vscode.window.onDidChangeTextEditorVisibleRanges,
+        vscode.window.onDidChangeTerminalState,
         vscode.window.onDidChangeTextEditorViewColumn,
+        vscode.workspace.onDidOpenTextDocument,
     ]
+
+    private register(disposable: vscode.Disposable) {
+        this.disposables.push(disposable)
+    }
+
+    /**
+     * Registers events that have conditions if they should even fire
+     */
+    private registerEventsWithEdgeCases(throttledEmit: (e: vscode.Event<any>) => any) {
```

Comment:
Could be just `registerAll`, so all cases live in one place. Separating them adds indirection without much benefit.